### ISSUE TITLE
Testing on Android

### ISSUE THREAD

**Issue by xyboox** [j3k0/cordova-plugin-purchase#574]
Hi,
I'm trying to test the IAP on android and everything works so far, except that if I use the signed version on my device I can't see the logs. Followed this instructions:
"To test locally, install the same same release signed app on a device which is logged into play store using a test account. If you use the debug build, you can do some actions, however you will not be able to buy the product."
Is there a way to see the logs while running the signed apk? Chrome inspector doesn't show the app in chrome://inspect/#devices
Thanks

**Comment by Renzo04** [comment]
You can "debug" your signed apk with javascript alerts, i don't know another way... Here's an example of my project code:

```
store.when(subscription.id).approved(function(p) {
  alert("Store approved");
  verifyTransaction(p).then(function() {
    p.finish();
  }).catch(function(error) {
    alert(angular.toJson(error));
  });
});

store.when(subscription.id).verified(function(p) {
  alert("Store verified");
  p.finish();
});

store.when(subscription.id).unverified(function(p) {
  alert("Store unverified");
});

store.when(subscription.id).updated(function(product) {
  if(product.loaded && product.valid && product.state === store.APPROVED) {
    product.finish();
  }
  alert("Store updated: " + product.state);
});
```

**Comment by xyboox** [comment]
Thanks @Renzo04 for the suggestion. This is very frustrating... How about logcat? Any chanse to filter out what we need from the flood of incoming logs over there?

**Comment by Renzo04** [comment]
Yes! Confirmed, you can debug your signed APK with logcat. You have to plug you device to your computer and in linux console run this command:

`adb logcat | grep -i "console"`

Source: [https://forum.ionicframework.com/t/js-console-log-on-logcat/17182/4](url)

Note: I'm working with ionic1

**Comment by xyboox** [comment]
Unfortunately, on Mac, I can only see this for every console.log message:

D/WebViewCallback(30228): onConsoleMessage
D/WebViewCallback(30228): onConsoleMessage
D/WebViewCallback(30228): onConsoleMessage
D/WebViewCallback(30228): onConsoleMessage
.......
D/WebViewCallback(30228): onConsoleMessage

Have you tested on some Linux distro and is showing actual console.log messages from the app, @Renzo04 ?

**Comment by Renzo04** [comment]
Mmm... yes, my linux distro is Ubuntu 16.04, and the console messages from the app are shown, this is an example:

```
04-06 12:53:22.586  9517  9517 I chromium: [INFO:CONSOLE(29)] "some message", source: http://10.0.0.40:8100/?restart=488568 (29)
04-06 12:53:22.588  9517  9517 I chromium: [INFO:CONSOLE(29)] "some message", source...
```

**Comment by xyboox** [comment]
using console.log or console.info in app? Does it even matters?

**Comment by Renzo04** [comment]
Because i'm using ionic (angularJS), instead console.log i used $log.log, but i think it doesn't matters, you can check this solution and try:

[Stackoverflow forums](http://stackoverflow.com/questions/25610936/enable-logcat-on-release-build-in-android-studio)

**Comment by xyboox** [comment]
I'm also using Ionic ( so angular ). I'll give it a try with $log.log as well. Maybe...

**Comment by Renzo04** [comment]
Remember to inject $log service as dependency inside your controller, factory, service, etc.

**Comment by albertleao** [comment]
Hey all, I just got this working so I can debug my 'release' version of the app and test the in-app-billing. This is what I did.

- Upload my build to Alpha/Beta channel of google play
- Open the cordova project in Android Studio
- add the following to your application node in the android manifest :  `android:debuggable="true"`
- Build a new apk with Android Studio
- Deploy to your usb device using ADB install command `adb -d install /path/to/release.apk`

It loaded the release app on my physical device and I am still able to inspect the app via chrome. Hope this helps.
